library/init/data/nat/lemmas.lean

```diff
@@ -614,8 +614,27 @@ nat.strong_induction_on a $ λ n,
   end
 
 /- mod -/
+
+private lemma mod_core_congr {x y f1 f2} (h1 : x ≤ f1) (h2 : x ≤ f2) :
```

Comment:
no private please

Reply:
I'm happy to make it protected instead, but I vehemently disagree with the claim that it's useful.  You're not supposed to use `nat.mod_core`, ever.